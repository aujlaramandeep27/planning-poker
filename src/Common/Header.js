import React from 'react';

export function withHeader(WrappedComponent, title) {
  return ({goBack, ...props}) => (
    <div className="default-background">
      <div className="uk-container uk-text-center@m">
        <h1 className="uk-heading-divider uk-margin-top">{title}</h1>
        <div className="uk-align-center uk-width-1-2@m">
          <span data-uk-icon="icon: arrow-left; ratio: 3" className="uk-position-large uk-position-top-left" onClick={goBack}/>
          <WrappedComponent {...props}/>
        </div>
      </div>
    </div>
  )
}
