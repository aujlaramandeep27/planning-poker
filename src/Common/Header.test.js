import { noop } from 'lodash';
import { withHeader } from './Header';
import React from 'react';

describe('Higher Order Component: withHeader', () => {
  function FakeComponent() {
    return <div/>
  }

  it('should render a Header on a component correctly', () => {
    expect(withHeader(FakeComponent, 'Fake Component')({ onBack: noop })).toMatchSnapshot();
  })
});
