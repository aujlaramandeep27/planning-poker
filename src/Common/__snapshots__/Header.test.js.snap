// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Higher Order Component: withHeader should render a Header on a component correctly 1`] = `
<div
  className="default-background"
>
  <div
    className="uk-container uk-text-center@m"
  >
    <h1
      className="uk-heading-divider uk-margin-top"
    >
      Fake Component
    </h1>
    <div
      className="uk-align-center uk-width-1-2@m"
    >
      <BackButton />
      <FakeComponent
        onBack={[Function]}
      />
    </div>
  </div>
</div>
`;
